packages/cli/src/helpers/errors.ts

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import chalk from 'chalk';
+
+export class ExitCodeError extends Error {
+  readonly code: number;
+
+  constructor(code: number, command?: string) {
+    if (command) {
+      super(`Command '${command}' exited with code ${code}`);
+    } else {
+      super(`Child exited with code ${code}`);
+    }
+    this.code = code;
+  }
+}
+
+export function isExitCodeError(error: Error): error is ExitCodeError {
+  return error.name === 'ExitCodeError';
```

Comment:
can one do like `error.constructor === ExitCodeError` instead?

Reply:
Yeah got rid of the name check since we're not really expecting any kind of deserialized errors. Added a CustomError base class to capture the name though.

In browser code I prefer checking by name though, bit more robust when you're not quite sure how errors are constructed